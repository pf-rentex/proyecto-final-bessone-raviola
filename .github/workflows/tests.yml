name: Test suites

on: [push]

jobs:
  Frontend:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Front install deps
        working-directory: ./frontend
        run: npm install
      - name: run tests
        working-directory: ./frontend
        run: npm run test

  Backend:
    runs-on: ubuntu-latest
      steps:
        - uses: actions/checkout@v1
        - name: Install deps
          working-directory: ./backend
          run: npm install
        - name: run tests
          working-directory: ./backend
          run: npm run test
