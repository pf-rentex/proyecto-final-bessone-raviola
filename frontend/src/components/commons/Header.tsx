import React from 'react';
import { ReactComponent as Logo } from "../../assets/logo.svg";

const Header = (props: { 
    //buttonRegistrer: boolean,
}) => {

    return (
        <div>
            <Logo />
        </div>
    );

};

export default Header;