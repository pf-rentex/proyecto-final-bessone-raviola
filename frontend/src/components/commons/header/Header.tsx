import React, { useEffect } from 'react';
import { GiHamburgerMenu } from 'react-icons/gi';
import { ReactComponent as LogoHeader } from '../../../assets/logo_header.svg';
import { ReactComponent as LogoHeaderMbl } from '../../../assets/favicon.svg';
import CustomButton from '../Button/CustomButton';
import { useNavigate } from 'react-router-dom';
import { connect } from 'react-redux';
import { loadUser, logout } from '../../../actions/auth';
import { FaBell } from 'react-icons/all';

interface IHeaderProps {
    loadUser: () => void;
    isAuthenticated: boolean;
    setIsOpenSidebar: Function;
    toggleNotifications: Function;
    logout: () => void;
}

const Header = (
    { loadUser, isAuthenticated, setIsOpenSidebar, toggleNotifications, logout }: IHeaderProps,
    history: History,
) => {
    const navigate = useNavigate();
    useEffect(() => {
        loadUser();
    }, [loadUser]);

    return (
        <div
            style={{ backgroundColor: '#20323A' }}
            className="flex flex-row sticky top-0 z-20 w-full items-center h-20 overflow-hidden"
        >
            <div className="flex flex-shrink items-center">
                {isAuthenticated && (
                    <div className="ml-5">
                        <GiHamburgerMenu
                            data-testid="sidebar-toggle"
                            className="text-white text-3xl cursor-pointer"
                            onClick={() => setIsOpenSidebar(true)}
                        />
                    </div>
                )}
                <div className="ml-6 cursor-pointer" onClick={() => navigate('/')}>
                    <LogoHeader className="w-40 mx-4 md:flex hidden" data-testid="logo" />
                    <LogoHeaderMbl className="visible flex md:hidden w-10 mx-4" />
                </div>
            </div>

            <div className="flex flex-1 justify-end items-center">
                {!isAuthenticated && (
                    <>
                        <div className="mr-4">
                            <button className="text-xs md:text-base text-white">
                                <u>Sobre nosotros</u>
                            </button>
                        </div>
                        <div className="ml-6 mr-7">
                            <CustomButton text="Registrarse" callback={() => navigate('login')} />
                        </div>
                    </>
                )}
                {isAuthenticated && (
                    <div className="flex flex-1 justify-end">
                        <button className="mx-3" onClick={() => toggleNotifications()}>
                            <FaBell color="white" size={25} />
                        </button>
                        <button className="mx-3" onClick={() => navigate('/profile')}>
                            {/* Se reemplazaria por la foto de perfil del usuario*/}
                            <img
                                src="https://creazilla-store.fra1.digitaloceanspaces.com/emojis/43191/blue-circle-emoji-clipart-md.png"
                                width="50px"
                                alt="Profile"
                            />
                        </button>
                        {/*    TODO: Dropdown ? */}
                        <button className="text-white mx-3" onClick={logout}>
                            Cerrar sesi√≥n
                        </button>
                    </div>
                )}
            </div>
        </div>
    );
};

const mapStateToProps = (state: any) => ({
    isAuthenticated: state.auth.isAuthenticated,
});

export default connect(mapStateToProps, {
    loadUser,
    logout,
})(Header);
