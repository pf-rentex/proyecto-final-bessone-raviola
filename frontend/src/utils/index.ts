import { Ref, useEffect } from 'react';

export const genId = (prefix: string = '') => `${prefix}_${Math.random().toString(36).substr(2, 9)}`;

export const useClickedOutside = (ref: Ref<any>, callback: Function) => {
    useEffect(() => {
        const handleClickOutside = (event: Event) => {
            // @ts-ignore
            if (ref && ref.current && !ref.current.contains(event.target)) {
                callback();
            }
        };
        // Bind
        document.addEventListener('mousedown', handleClickOutside);
        return () => {
            // dispose
            document.removeEventListener('mousedown', handleClickOutside);
        };
        // @ts-ignore
    }, [ref, callback]);
};
