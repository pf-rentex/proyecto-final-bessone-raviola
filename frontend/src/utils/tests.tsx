import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

const middlewares = [thunk];

const useMockedReduxState = (state: Object, children: JSX.Element) => {
    const mockStore = configureMockStore(middlewares);
    const store = mockStore(state);
    const wrapper = render(<Provider store={store}>{children}</Provider>);

    return wrapper;
};

export { useMockedReduxState };
