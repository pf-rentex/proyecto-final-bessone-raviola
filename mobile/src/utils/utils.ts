export const genId = (prefix: string) =>
  `${prefix}_${Math.random().toString(36).substr(2, 9)}`;
