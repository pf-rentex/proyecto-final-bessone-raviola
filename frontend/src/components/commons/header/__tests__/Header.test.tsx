import React from 'react';
import { cleanup, fireEvent, getAllByTestId, render } from '@testing-library/react';
import Header from '../Header';
import { useMockedReduxState } from '../../../../utils/tests';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));

describe('Header test cases', () => {
    let wrapper: any;

    const renderComponent = (component: JSX.Element) => {
        const mockedState = {
            auth: {
                isAuthenticated: true,
            },
        };
        // eslint-disable-next-line react-hooks/rules-of-hooks
        return useMockedReduxState(mockedState, component);
    };

    beforeEach(() => {
        wrapper = renderComponent(<Header setIsOpenSidebar={() => {}} />);
    });
    afterEach(cleanup);

    test('renders component', () => {
        // const wrapper = render(<Header setIsOpenSidebar={() => {}} />);

        expect(wrapper.container).toBeInTheDocument();
        expect(wrapper.container).toMatchSnapshot();
    });

    test('It contains the App logo', () => {
        const { getByTestId } = wrapper;

        expect(getByTestId('logo')).toBeInTheDocument();
    });

    test('It contains the sidebar toggle', () => {
        const { getByTestId } = wrapper;

        expect(getByTestId('sidebar-toggle')).toBeInTheDocument();
    });

    test('callback for setIsOpenSidebar is called', () => {
        const callback = jest.fn();
        const { getAllByTestId } = renderComponent(<Header setIsOpenSidebar={callback} />);

        getAllByTestId('sidebar-toggle').map((element: any) => {
            fireEvent.click(element);
        });

        expect(callback).toHaveBeenCalled();
    });

    test('Contains the button to logout', () => {
        const { getByText } = wrapper;

        getByText('Cerrar sesiÃ³n');
    });

    test('Contains the Register button', () => {
        const { getByText } = useMockedReduxState(
            {
                auth: {
                    isAuthenticated: false,
                },
            },
            <Header setIsOpenSidebar={() => {}} />,
        );

        getByText('Registrarse');
    });
});
