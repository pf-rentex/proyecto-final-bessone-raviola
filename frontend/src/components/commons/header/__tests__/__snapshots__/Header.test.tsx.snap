// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test cases renders component 1`] = `
<div>
  <div
    class="flex flex-row sticky top-0 z-20 w-full items-center h-20 overflow-hidden"
    style="background-color: rgb(32, 50, 58);"
  >
    <div
      class="flex flex-shrink items-center"
    >
      <div
        class="ml-5"
      >
        <svg
          class="text-white text-3xl cursor-pointer"
          data-testid="sidebar-toggle"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
      </div>
      <div
        class="ml-6"
      >
        <svg
          class="w-40 mx-4 md:flex hidden"
          data-testid="logo"
        >
          logo_header.svg
        </svg>
        <svg
          class="visible flex md:hidden w-10 mx-4"
        >
          favicon.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 justify-end items-center"
    >
      <div
        class="mr-4"
      >
        <a
          class="text-xs md:text-base text-white"
          href="#"
        >
          <u>
            Sobre nosotros
          </u>
        </a>
      </div>
      <div
        class="ml-6 mr-7"
      >
        <button
          class="bg-primary  shadow-lg w-full rounded-md px-5 py-2 flex content-center my-3 focus:shadow-sm font-secondary font-semibold outline-none align-items-center justify-center content-center align-middle justify-items-center items-center"
        >
          <span
            class="text-2xl pr-4"
          />
          <span
            class="text-sm font-sm  text-white flex-1"
          >
            Registrarse
          </span>
        </button>
      </div>
      <div
        class="mr-5"
      />
    </div>
  </div>
</div>
`;
